### DISCUSSION TITLE
How do I search the File Library for items with a certain tag?

### DISCUSSION THREAD

**Discussion by AaronBeaudoin** [directus/directus#13418] [Archive]
I typically think of tags as "keywords", a way to identify items for quick search later on. So I've tagged a few files in Directus and tried to search for files using the value of a tag... and... nothing.

I also tried adding a filter and the only options are "Is null" and "Isn't null". What?

<img width="527" alt="image" src="https://user-images.githubusercontent.com/26592486/169354595-79f9bb29-8935-4786-9e0b-511346b841a6.png">

Can I search for files using tags? If so, how? If not, then what should I be using tags for instead? And what should I use for searching with "keywords"? Thanks!

**Comment by benhaynes** [comment]
Have you tried the `in` operator? That is ideal for array types.

**Reply by AaronBeaudoin** [reply]
Not sure what you mean. I don't see an option for an `in` operator, and of course "{value} in Tags" is just interpreted as plain text. Can you show an example?

Also how do I know the "Tags" field is even an array type? It's not as if I created the field, as the files collection is built-in.

**Reply by benhaynes** [reply]
You're right! Within the File library we have `tags` and I only see the two (NULL and NOT NULL) operators for that field. This seems to be missing options, right @rijkvanzanten ?

<img width="204" alt="Screen Shot 2022-05-19 at 12 42 00 PM" src="https://user-images.githubusercontent.com/522079/169376529-455e1d6d-d2d1-4a50-9591-e7c8eccbd4fd.png">

**Reply by AaronBeaudoin** [reply]
I just found #12496, but I'm confused because I don't understand why file items even have a "Tags" field if you can't search by it. Like, isn't the whole purpose of a "Tags" field to be searched?

**Reply by benhaynes** [reply]
The ticket you referenced is talking about _search_. However, here he's operators are part of _filtering_... Two distinct capabilities. I agree with you that filtering should support proper array operators, this discussion will likely be converted into an issue to track the appropriately.

**Comment by piotr-cz** [comment]
I think the more details about the issue are in this discussion: https://github.com/directus/directus/discussions/7277
